python/crusher.py

```diff
@@ -0,0 +1,163 @@
+krylov_params["monitor_convergence"] = False
+krylov_params["maximum_iterations"] = 15000
+
+processID =  MPI.comm_world.Get_rank()
+set_log_level(50) #50, 40, 30, 20, 16, 13, 10
+
+foldername = "crusher_2w/"
+t_end = 1.05#8
+dt = 1./1000. #1600
+t=0.
+n=0
+
+fname = "geo/crusher_2w"
+mesh = Mesh(fname + '.xml')
+cells = MeshFunction('size_t', mesh , fname + '_physical_region.xml')
+facets = MeshFunction('size_t', mesh , fname + '_facet_region.xml')
+#nn = FacetNormal(mesh)
+#on the server benz, the xml cannot be read, How to deal with this problem?
+
+i, j, k, l = indices(4)
+delta = Identity(2)
+
+scalar = FiniteElement('P', triangle, 1)
+vector = VectorElement('P', triangle, 1)
+mixed_element = MixedElement([scalar, scalar, vector])#p,J,v
+Space = FunctionSpace(mesh, mixed_element)
+
+da = Measure( 'ds' , domain=mesh , subdomain_data=facets )
+dv = Measure( 'dx' , domain=mesh , subdomain_data=cells )
+
+# Define the values of dimensionless quantities
+g_=0.907
+a_=1.575
+J_up=20.
+J1=1.
+P_up=0.
+L=0.5
+H=1.
+lambd=0.125
+mu0=0.01
+mu1=0.1
+
+bc = []
+# Define boundary conditions
+bc.append( DirichletBC(Space.sub(1), J_up,facets, 2) )
+bc.append( DirichletBC(Space.sub(2), Constant((0.,0.)), facets, 1) )
+bc.append( DirichletBC(Space.sub(0), P_up, facets,2) )
+bc.append( DirichletBC(Space.sub(0), 0., facets,3) )
+
+dunkn = TrialFunction(Space)
+test = TestFunction(Space)
+unkn = Function(Space)
+unkn0 = Function(Space)
+unkn00 = Function(Space)
+
+'''
+file_p = XDMFFile(foldername+'/p.xdmf')
+file_J = XDMFFile(foldername+'/J.xdmf')
+file_v = XDMFFile(foldername+'/v.xdmf')
+p_out, J_out, v_out = unkn.split(deepcopy=True)
+file_p.write_checkpoint(p_out, "p", t, append=False)
+file_J.write_checkpoint(J_out, "J", t, append=False)
+file_v.write_checkpoint(v_out, "v", t, append=False)
+'''
+file_p = File(foldername+'p.pvd')
+file_J = File(foldername+'J.pvd')
+file_v = File(foldername+'v.pvd')
+'''
+The xdmf is better, becasue the result only have two files, .pvd stores the results of each step separately. 
+But the xdmf result sovled by my computer cannot open by paraview 
+(I try to open it by paraview5.6 and 5.9 under MacOS and Ubuntu, 
+ none of these work. maybe because of the version of the annaconda, My fenics is built under anaconda)
+The xdmf result files from server benz is ok, but here, I use the .xml 
+and when I run this code on benz, it will report an error
+
+'''
+
+p0, J0, v0 = split(unkn0)
+p, J, v = split(unkn)
+del_p, del_J, del_v = split(test)
+
+
+#dimensionless quantities
+
+mu = mu0 + (mu1 - mu0)*exp(-lambd*(J-J1)/J1)
+d = as_tensor( 1./2.*(v[i].dx(j)+v[j].dx(i)) , [i,j] )
+tau = as_tensor( 2.*mu*d[i,j] , [i,j] )
+#tau = as_tensor( mu*v[i].dx(j) , [i,j] )
+g = Expression(('0.','-g'),degree=1,g=0.907)
+F_1 = v[i].dx(i)*del_p*dv
+F_2 = ( ( (v-v0)[j]/dt -g[j] + (v[i]*v[j]).dx(i) + p.dx(j) )*del_v[j]  + tau[i,j]*del_v[j].dx(i) )*dv 
+F_3 = ( (J-J0)/dt + v[i]*J.dx(i) + a_*p*(J-1) )*del_J*dv
```

Comment:
I don't think that you can discretize the evolution equation for the inertia tensor in this way. It is known in the literature that a transport equation is numerically unstable and requires some special treatment.

Reply:
Yes, I have no confidence in this part, I just used the equation directly after generating a weak form